Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs

```diff
@@ -0,0 +1,238 @@
+                string.Empty,
+                executionContext.TimeLimit,
+                executionContext.MemoryLimit,
+                arguments);
+
+            var errorsByFiles = this.GetTestErrors(processExecutionResult.ReceivedOutput);
+            var testIndex = 0;
+
+            foreach (var test in executionContext.Tests)
+            {
+                // Construct and figure out what the Test results are
+                TestResult testResult = new TestResult()
+                {
+                    Id = test.Id,
+                    TimeUsed = (int)processExecutionResult.TimeWorked.TotalMilliseconds,
+                    MemoryUsed = (int)processExecutionResult.MemoryUsed
+                };
+
+                switch (processExecutionResult.Type)
+                {
+                    case ProcessExecutionResultType.RunTimeError:
+                        testResult.ResultType = TestRunResultType.RunTimeError;
+                        testResult.ExecutionComment = processExecutionResult.ErrorOutput.MaxLength(2048); // Trimming long error texts
+                        break;
+                    case ProcessExecutionResultType.TimeLimit:
+                        testResult.ResultType = TestRunResultType.TimeLimit;
+                        break;
+                    case ProcessExecutionResultType.MemoryLimit:
+                        testResult.ResultType = TestRunResultType.MemoryLimit;
+                        break;
+                    case ProcessExecutionResultType.Success:
+                        string message = "Test Passed!";
+                        string testFile = this.testNames[testIndex++];
+                        if (errorsByFiles.ContainsKey(testFile))
+                        {
+                            message = errorsByFiles[testFile];
+                        }
+
+                        var checkerResult = checker.Check(test.Input, message, test.Output, test.IsTrialTest);
+                        testResult.ResultType = checkerResult.IsCorrect ? TestRunResultType.CorrectAnswer : TestRunResultType.WrongAnswer;
+                        testResult.CheckerDetails = checkerResult.CheckerDetails;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(processExecutionResult), "Invalid ProcessExecutionResultType value.");
+                }
+
+                result.TestResults.Add(testResult);
+            }         
+
+            return result;
+        }
+
+        private Dictionary<string, string> GetTestErrors(string receivedOutput)
+        {
+            var errorsByFiles = new Dictionary<string, string>();
+            var errorRegex =
+     new Regex($@"\d+\) (.*){Environment.NewLine}((?:.+{Environment.NewLine})*?)\s*at (?:[^(){Environment.NewLine}]+?)\(\) in \w:\\(?:[^\\{Environment.NewLine}]+\\)*(.+).cs");
+
+            var errors = errorRegex.Matches(receivedOutput);
+
+            foreach (Match error in errors)
```

Comment:
It's a bug in C#, foreaching a MatchCollection yields Objects instead of Matches, so an explicit cast is required.